test/e2e/providers/azure/azure.go

```diff
@@ -119,11 +124,218 @@ func (p *Provider) GetType() config.PlatformType {
+								},
+							},
+						},
+						{
+							Name:  "azurefile",
+							Image: "mcr.microsoft.com/k8s/csi/azurefile-csi:latest",
+							Args:  []string{"--v=5", "--endpoint=$(CSI_ENDPOINT)", "--nodeid=$(KUBE_NODE_NAME)"},
+							Env: []core.EnvVar{
+								{
+									Name: "KUBE_NODE_NAME",
+									ValueFrom: &core.EnvVarSource{
+										FieldRef: &core.ObjectFieldSelector{
+											FieldPath: "spec.nodeName",
+										},
+									},
+								},
+								{
+									Name:  "CSI_ENDPOINT",
+									Value: `unix://C:\\csi\\csi.sock`,
+								},
+							},
+							VolumeMounts: []core.VolumeMount{
+								{
+									Name:      "kubelet-dir",
+									MountPath: "/var/lib/kubelet",
+								},
+								{
+									Name:      "plugin-dir",
+									MountPath: "/csi",
+								},
+								{
+									Name:      "azure-config",
+									MountPath: "/k",
+								},
+								{
+									Name:      "csi-proxy-filesystem-v1",
+									MountPath: `\\.\pipe\csi-proxy-filesystem-v1`,
+								},
+								{
+									Name:      "csi-proxy-smb-pipe-v1",
+									MountPath: `\\.\pipe\csi-proxy-smb-v1`,
+								},
+							},
+						},
+					},
+					Volumes: []core.Volume{
+						{
+							Name: "csi-proxy-filesystem-v1",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `\\.\pipe\csi-proxy-filesystem-v1`,
+								},
+							},
+						},
+						{
+							Name: "csi-proxy-smb-pipe-v1",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `\\.\pipe\csi-proxy-smb-v1`,
+								},
+							},
+						},
+						{
+							Name: "registration-dir",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `C:\var\lib\kubelet\plugins_registry\`,
+									Type: &directoryType,
+								},
+							},
+						},
+						{
+							Name: "kubelet-dir",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `C:\var\lib\kubelet`,
+									Type: &directoryType,
+								},
+							},
+						},
+						{
+							Name: "plugin-dir",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `C:\var\lib\kubelet\plugins\file.csi.azure.com\`,
+									Type: &directoryOrCreate,
+								},
+							},
+						},
+						{
+							Name: "azure-config",
+							VolumeSource: core.VolumeSource{
+								HostPath: &core.HostPathVolumeSource{
+									Path: `C:\k\`,
+									Type: &directoryOrCreate,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	// See if the DaemonSet already exists in the state we expect it to be in.
+	existingDS, err := client.AppsV1().DaemonSets(csiNamespace).Get(context.TODO(), dsName, meta.GetOptions{})
+	if err != nil {
+		if !k8sapierrors.IsNotFound(err) {
+			return fmt.Errorf("error getting existing Windows CSI DaemonSet: %w", err)
+		}
+	} else if err == nil {
```

Comment:
No need for `if err == nil`

Reply:
It is needed in order to exclude the case where `k8sapierrors.IsNotFound(err)` evaluates to true

Reply:
That case is excluded by the `else` statement. `err == nil` is always true in the block following the `else` in
```
if err != nil {
	
} else {

}
```

Reply:
yep you are right, fixed